# Origin Analysis

The determination of the most recent common ancestor of the human SARS-CoV-2
and the zoonotic transmission can provide important information about biological features, key
mutations and properties of the virus.

This project is based on Comparing the genetic sequence of SARS-CoV-2 with other viruses of the Coronaviridae
family in different hosts. Six complete genomes were considered, The sequence data (fasta files) were downloaded from the NCBI Virus public library.

The phylogenetic computations are performed by the hierarchical UPGMA and the TreeTime algorithm
and the results are compared and analyzed afterwards. One top of that two non-hierarchical
clustering methods (k-means and k-medoids) are performed to compare its results with the outcomes
of the hierarchical clustering.
